specs/commandline/verify.md

```diff
@@ -168,3 +169,23 @@ An example of output messages for a successful verification:
 Warning:  Always verify the artifact using digest(@sha256:...) rather than a tag(:v1) because resolved digest may not point to the same signed artifact, as tags are mutable.
 Successfully verified signature for localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
 ```
+
+### Verify signatures on an OCI artifact with json output
+
+Use the `--output` flag to format successful verification output in json.
+
+```shell
+notation verify --output json localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+```
+
+An example of output messages for a successful verification:
+
+```text
+{
+    "reference": "localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9",
+    "userMetadata": {
+        "io.wabbit-networks.buildId": "123"
+    },
+    "result": "Success"
```

Comment:
for verification failure,  no json is written to stdout, and the failure is logged to stderr.

Reply:
@byronchien If JSON is printed only if the verification passes, what's the meaning of showing `"result": "Success"`?

/cc @priteshbandi @yizha1

Reply:
When verification fails, I expect that the result would be `failure` or `failed`, and the failure reason will be included in the JSON object so that other scripts can parse it correctly.

Reply:
Since this PR is merged, I've created https://github.com/notaryproject/notation/issues/546 to track.

Reply:
`result` can have two values, one is `skip` and `success`. IMO for failure case displaying JSON for wont be useful because it wont contain any useful information which automation/script can use(non-zero exit code should suffice to show failure)

Reply:
The JSON output is usually consumed by scripts or programs. How can them obtain the structured error message and detailed verification outcomes?

Reply:
To display error message/detailed verification output we need more than `result` field something like `resultReason` because `result` will only say `failure`. Also, in current state it wont be of much help since as there is only one genuine/expected failure use case i.e failed signature verification for all the signatures. Apart from genuine/expected error, all other errors should always be emitted as stdErr.

Are you suggest we should emit json for expected failure use cases or all failure use cases?